src/llir/intrinsic.rs

```diff
@@ -254,18 +252,8 @@ impl IntrinsicAbiHelper<'_> {
         }
     }
 
-    fn find_and_remove_padding(&self, arg_encodings: &mut Vec<(usize, &ArgEncoding)>) -> Result<abi_parts::PaddingInfo, Diagnostic> {
-        let mut count = 0;
-        let mut first_index = arg_encodings.len();
-        while let Some(&(index, ArgEncoding::Padding)) = arg_encodings.last() {
-            // assumption that this func always runs first (nothing is deleted before us)
-            assert_eq!(index, arg_encodings.len() - 1);
-
-            arg_encodings.pop();
-            count += 1;
-            first_index = index;
-        }
-        Ok(abi_parts::PaddingInfo { count, index: first_index })
+    fn find_and_remove_padding(&self, arg_encodings: &mut Vec<(usize, &ArgEncoding)>) {
+        arg_encodings.retain(|(_, enc)| !matches!(*enc, ArgEncoding::Padding { .. }));
```

Comment:
This was also simplified by changing padding to be format letters instead of optional args